content/api/module.md

```diff
@@ -0,0 +1,399 @@
+``` javascript
+var a = require('normal-dep');
+
+if ( /* Some Condition */ ) {
+  require.ensure(["b"], function(require) {
+    var c = require("c");
+
+    // Do something special...
+  });
+}
+```
+
+
+
+## AMD
+
+Asynchronous Module Definition (AMD) is a JavaScript specification that defines an interface for writing and loading modules. The following AMD methods are supported by webpack:
+
+
+### `define` (with factory)
+
+``` javascript
+define([name: String], [dependencies: String[]], factoryMethod: function(...))
+```
+
+If `dependencies` are provided, `factoryMethod` will be called with the exports of each dependency (in the same order). If `dependencies` are not provided, `factoryMethod` is called with `require`, `exports` and `module` (for compatibility!). If this function returns a value, this value is exported by the module. The compiler ensures that each dependency is available.
+
+W> Note that webpack ignores the `name` argument.
+
+``` javascript
+define(['jquery', 'my-module'], function($, myModule) {
+  // Do something with $ and myModule...
+
+  // Export a function
+  return function doSomething() {
+    // ...
+  };
+});
+```
+
+W> This CANNOT be used in an asynchronous function.
+
+
+### `define` (with value)
+
+``` javascript
+define(value: !Function)
+```
+
+This will simply export the provided `value`. The `value` here can be anything except a function.
+
+``` javascript
+define({
+  answer: 42
+});
+```
+
+W> This CANNOT be used in an async function.
+
+
+### `require`
+
+``` javascript
+require(dependencies: String[], [callback: function(...)])
+```
+
+Similar to `require.ensure`, this will split the given `dependencies` into a separate bundle that will be loaded asynchronously. The `callback` will be called with the exports of each dependency in the `dependencies` array.
+
+``` javascript
+require(['b'], function(b) {
+  var c = require("c");
+});
+```
+
+W> There is no option to provide a chunk name.
+
+
+
+## Labeled Modules
+
+The internal `LabeledModulesPlugin` enables you to use the following methods for exporting and requiring within your modules:
+
+
+### `export` label
+
+Export the given `value`. The label can occur before a function declaration or a variable declaration. The function name or variable name is the identifier under which the value is exported.
+
+``` javascript
+export: var answer = 42;
+export: function method(value) {
+  // Do something
+};
+```
+
+W> Using it in an async function may not have the expected effect.
+
+
+### `require` label
+
+Make all exports from the dependency available in the current scope. The `require` label can occur before a string. The dependency must export values with the `export` label. CommonJS or AMD modules cannot be consumed.
+
+__some-dependency.js__
+
+``` javascript
+export: var answer = 42;
+export: function method(value) {
+  // Do something...
+};
+```
+
+``` javascript
+require: 'some-dependency';
+console.log(answer);
+method(...);
+```
+
+
+
+## Webpack
```

Comment:
I think maybe it would make more sense to split at the `## Module Variables` section? Until we have that a nicer page grouping (#438) maybe something like `module-api-methods.md` and `module-api-variables.md`?